src/uucore/src/lib/features/utmpx.rs

```diff
@@ -345,13 +376,137 @@ impl UtmpxIter {
         Self {
             guard,
             phantom: PhantomData,
+            #[cfg(feature = "feat_systemd_logind")]
+            systemd_iter: None,
+        }
+    }
+
+    #[cfg(feature = "feat_systemd_logind")]
+    fn new_systemd() -> Self {
+        // PoisonErrors can safely be ignored
+        let guard = LOCK.lock().unwrap_or_else(|err| err.into_inner());
+        let systemd_iter = systemd_logind::SystemdUtmpxIter::new().ok();
+        Self {
+            guard,
+            phantom: PhantomData,
+            systemd_iter,
+        }
+    }
+}
+
+/// Wrapper type that can hold either traditional utmpx records or systemd records
+pub enum UtmpxRecord {
+    Traditional(Box<Utmpx>),
+    #[cfg(feature = "feat_systemd_logind")]
+    Systemd(systemd_logind::SystemdUtmpxCompat),
+}
```

Comment:
I'm a bit confused about this enum and it's implementation. What's the reason for solving it in this way? Why not define a trait `UtmpxRecord` and implement it for both `Utmpx` and `SystemdUtmpxCompat`? Probably I'm missing something.

Reply:
Yes, you are right, I didn't think about the memory allocation. I was thinking on a higher level, with memory being a black box.